tests/ml/test_ml_model_pytest.py

```diff
@@ -306,6 +319,98 @@ def test_decision_tree_regressor(self, compress_model_definition):
+        regressor = DecisionTreeRegressor()
+        regressor.fit(training_data[0], training_data[1])
+
+        # Serialise the models to Elasticsearch
+        model_id = "test_learning_to_rank"
+        feature_extractors = [
+            {
+                "query_extractor": {
+                    "feature_name": "title_bm25",
+                    "query": {"match": {"title": "{{query_string}}"}},
+                }
+            },
+            {
+                "query_extractor": {
+                    "feature_name": "imdb_rating",
+                    "query": {
+                        "script_score": {
+                            "query": {"exists": {"field": "imdbRating"}},
+                            "script": {"source": 'return doc["imdbRating"].value;'},
+                        }
+                    },
+                }
+            },
+        ]
+        feature_names = [
+            extractor["query_extractor"]["feature_name"]
+            for extractor in feature_extractors
+        ]
+        inference_config = {
+            "learning_to_rank": {"feature_extractors": feature_extractors}
+        }
+
+        es_model = MLModel.import_model(
+            ES_TEST_CLIENT,
+            model_id,
+            regressor,
+            feature_names,
+            es_if_exists="replace",
+            es_compress_model_definition=compress_model_definition,
+            inference_config=inference_config,
+        )
+
+        # Verify the saved inference config contains the passed LTR config
+        response = ES_TEST_CLIENT.ml.get_trained_models(model_id=model_id)
+        assert response.meta.status == 200
+        assert response.body["count"] == 1
+        saved_inference_config = response.body["trained_model_configs"][0][
+            "inference_config"
+        ]
+        assert "learning_to_rank" in saved_inference_config
+        saved_ltr_config = saved_inference_config["learning_to_rank"]
+        assert all(
+            item in saved_ltr_config.items()
+            for item in inference_config["learning_to_rank"].items()
+        )
+
+        # Execute search with rescoring
+        search_result = ES_TEST_CLIENT.search(
+            index=MOVIES_INDEX_NAME,
+            query={"terms": {"_id": ["tt1318514", "tt0071562"]}},
+            rescore={
+                "learning_to_rank": {
+                    "model_id": model_id,
+                    "params": {"query_string": "planet of the apes"},
+                }
+            },
+        )
+
+        # Assert that:
+        # - all documents from the query are present
+        # - all documents have been rescored (score != 1.0)
+        # - document scores are unique
+        doc_scores = [hit["_score"] for hit in search_result["hits"]["hits"]]
+        assert len(search_result["hits"]["hits"]) == 2
+        assert all(score != float(1) for score in doc_scores)
+        assert all(doc_scores.count(score) == 1 for score in doc_scores)
```

Comment:
@afoucret Looks like document scores are not unique. Among the two consequent runs of this test (`compress_model_definition=True|False`) I tend to get equal scores for the two documents in the hit list:
```
doc_scores [147.05865, 109.10927] # run 1
doc_scores [139.4883, 139.4883] # run 2 - this assertion fails 
```

I don't understand where the randomness comes from; shouldn't the results be deterministic with a static test fixture and LTR configuration?

Reply:
The randomness comes from the model which is trained from random data. Let me see if I can find a way to workaround this.

Reply:
For your PR, it seems impossible to control the randomness in the model. I think we should remove the tests that check the score are all different for now:
assert all(doc_scores.count(score) == 1 for score in doc_scores)
When I will add the feature logger, I will be able to

Reply:
Ok, removed.